nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/model/Property.java

```diff
@@ -500,6 +500,24 @@
      */
     Property get(int index) throws PropertyNotFoundException;
 
+    /**
+     * Sets the child property having the given name.
+     * <p>
+     * The given name should be the full name (i.e. prefixed name if any prefix exists).
+     * <p>
+     * If a non prefixed name is given, the first child property having the given local name will be returned.
+     * <p>
+     * Relative paths are not resolved. This method is intended to lookup direct children. For path lookups use
+     * {@link Property#resolvePath(String)} instead.
+     *
+     * @param name the child property name (the full name including the prefix if any)
+     * @param property the child property to set
+     * @throws UnsupportedOperationException if the current property is a scalar property (doesn't have children)
+     * @throws PropertyNotFoundException if the child property is not found in the type definition
+     * @since 11.1
+     */
+    void set(String name, Property property) throws PropertyException;
+
```

Comment:
i am wondering if these new methods on interface will not break any client code in a case where the code client implement the `Property` interface